src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,50 @@
 package core.basesyntax;
 
+abstract class Machine {
+    public abstract void doWork();
+    public abstract void stopWork();
+}
+class Truck extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Truck starts working");
+    }
+    @Override
+    public void stopWork() {
+        System.out.println("Truck stopped working");
+    }
+}
+class Bulldozer extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Bulldozer starts working");
+    }
+    @Override
+    public void stopWork() {
+        System.out.println("Bulldozer stopped working");
+    }
+}
+class Excavator extends Machine {
+    @Override
+    public void doWork() {
+        System.out.println("Excavator starts working");
```

Comment:
The message 'Excavator starts working' could be more informative. For example, it could indicate what kind of work the Excavator is doing.